scripts/wish/reduce.py

```diff
@@ -618,302 +421,182 @@ def WISH_process_incidentmon(number, ext, spline_terms=20, debug=False):
         else:
             n1, n2 = split_string(number)
             works = "monitor" + str(n1) + "_" + str(n2)
-            fname = WISH_getfilename(n1, ext)
+            filename = self.get_file_name(n1, ext)
             works1 = "monitor" + str(n1)
-            mantid.LoadRaw(Filename=fname, OutputWorkspace=works1, SpectrumMin=4, SpectrumMax=4, LoadLogFiles="0")
-            fname = WISH_getfilename(n2, ext)
+            mantid.LoadRaw(Filename=filename, OutputWorkspace=works1, SpectrumMin=4, SpectrumMax=4, LoadLogFiles="0")
+            filename = self.get_file_name(n2, ext)
             works2 = "monitor" + str(n2)
-            mantid.LoadRaw(Filename=fname, OutputWorkspace=works2, SpectrumMin=4, SpectrumMax=4, LoadLogFiles="0")
-            mantid.MergeRuns(InputWorkspaces=works1 + "," + works2, OutputWorkspace=works)
+            mantid.LoadRaw(Filename=filename, OutputWorkspace=works2, SpectrumMin=4, SpectrumMax=4, LoadLogFiles="0")
+            mantid.MergeRuns(InputWorkspaces=[works1, works2], OutputWorkspace=works)
             mantid.DeleteWorkspace(works1)
             mantid.DeleteWorkspace(works2)
             mantid.ConvertUnits(InputWorkspace=works, OutputWorkspace=works, Target="Wavelength", Emode="Elastic")
-        lmin, lmax = WISH_getlambdarange()
-        mantid.CropWorkspace(InputWorkspace=works, OutputWorkspace=works, XMin=lmin, XMax=lmax)
-        ex_regions = n.zeros((2, 4))
-        ex_regions[:, 0] = [4.57, 4.76]
-        ex_regions[:, 1] = [3.87, 4.12]
-        ex_regions[:, 2] = [2.75, 2.91]
-        ex_regions[:, 3] = [2.24, 2.50]
+        l_min, l_max = get_lambda_range()
+        mantid.CropWorkspace(InputWorkspace=works, OutputWorkspace=works, XMin=l_min, XMax=l_max)
```

Comment:
Any difference in speed?

Reply:
I'll do a full run to check, but it didn't seem to be anything appreciable

Reply:
the section I was testing with was on a single spectrum workspace, so not really a good example for the difference it might make (though it did shave ~2 seconds off the test), running elsewhere shows a slight increase in memory expense (~0.7 Gigabytes at max and maybe ~0.2 on average ) and actually slows the test back down by about 3 seconds. I'll leave in the one that sped up the test slightly

Reply:
Pete has pointed out that the algorithm takes a double list as arguments, so places where the original script had a whole block of `CropWorkspace` calls can be converted to a single call.

Reply:
I think i've already taken care of all of the blocks of cropworkspaces most of them were actually if else blocks where the cropping value depends on the panel, and crop ragged wont actually work because there not different spectrums in the same workspace, but entirely different workspaces

Reply:
There were some places in the original script that looked applicable. Such as here:

https://github.com/mantidproject/mantid/blob/6e3716dc9b07eb5e0e0ac082a921758662a16af6/scripts/wish/reduce.py#L400-L413

Reply:
I'm pretty sure that's actually a mistake, if you look above it it has exactly the same as for the individual panels above, but instead crops every panel to all of them, they've used panel zero to mean run every panel, so actually they've meant to crop each workspace to the cropping values for that panel, but have instead cropped every workspace to every cropping value, (which I believe in that case would just crop them all to the largest min and smallest max?but since I now crop immediately after focusing that block is removed anyway

Reply:
I see what you mean, yes sounds a little bit odd.

Reply:
There was a fair amount of little things like that, i.e. putting stuff in a loop but then doing every operation that the loop was meant to do inside the loop (they did it with saving as well, which made the whole thing take a huge amount of time longer than necessary) so it didn't seem that out of place a mistake

Reply:
Odd, put I'm not entirely surprised. But this is why I wanted a system test first, so when we change it we know if we broke it. Don't be afraid to add new tests if there is an area that isn't already covered.

Reply:
Yeah, I plan on adding one for the vanadium once I've verified that whats there matches up with the ones on auto reduction, it seems really off at the moment so I want to check it before i put a test in for it